panther_driver/src/panther_can.py

```diff
@@ -0,0 +1,173 @@
+        self._err_times = [0] * self._max_err_per_sec
+        
+        self._lock = Lock()
+        self._network = canopen.Network()
+
+        self._motor_controllers = [
+            MotorController(1, eds_file),   # front
+            MotorController(2, eds_file)    # rear
+        ]
+        
+        self._network.connect(channel=can_interface, bustype='socketcan')
+
+        self._network.add_node(self._motor_controllers[0].can_node)
+        self._network.add_node(self._motor_controllers[1].can_node)
+        
+        self._connection_check_timer = Thread(target=self._can_net_check)  
+        self._connection_check_timer.start()
+
+        rospy.loginfo(f'[{rospy.get_name()}] Connected to the CAN bus.')
+    
+    def is_can_connection_correct(self):
+        return self._can_net_err
+
+    def set_wheels_enc_velocity(self, vel: list) -> None:
+        with self._lock:
+            for motor_controller, enc_vel in zip(self._motor_controllers, [vel[:2], vel[2:]]):
+                for i, wheel in enumerate([LEFT_WHEEL, RIGHT_WHEEL]):
+                    try:
+                        motor_controller.can_node.sdo['Cmd_CANGO'][wheel].raw = enc_vel[i]
+                    except:
+                        rospy.logwarn(
+                            f'[{rospy.get_name()}] PantherCAN: SdoCommunicationError '
+                            f'occurred while setting wheels velocity'
+                        )
+                        self._error_handle()
+
+    def get_wheels_enc_pose(self):
+        with self._lock:
+            for motor_controller in self._motor_controllers:
+                for i, wheel in enumerate([LEFT_WHEEL, RIGHT_WHEEL]):
+                    try:
+                        motor_controller.wheel_pos[i] = motor_controller.can_node.sdo['Qry_ABCNTR'][wheel].raw
+                    except canopen.SdoCommunicationError:
+                        rospy.logwarn(
+                            f'[{rospy.get_name()}] PantherCAN: SdoCommunicationError ' 
+                            'occurred while reading wheels position'
+                        )
+                        self._error_handle()
+                    yield motor_controller.wheel_pos[i]
+
+    def get_wheels_enc_velocity(self):
+        with self._lock:
+            for motor_controller in self._motor_controllers:
+                for i, wheel in enumerate([LEFT_WHEEL, RIGHT_WHEEL]):
+                    try:
+                        motor_controller.wheel_vel[i] = motor_controller.can_node.sdo['Qry_ABSPEED'][wheel].raw
+                    except canopen.SdoCommunicationError:
+                        rospy.logwarn(
+                            f'[{rospy.get_name()}] PantherCAN: SdoCommunicationError '
+                            'occurred while reading wheels velocity'
+                        )
+                        self._error_handle()
+                    yield motor_controller.wheel_vel[i]
+
+    def get_motor_enc_current(self):
```

Comment:
```suggestion
    def get_motor_current(self):
```

You say encoders measure current in amperes?